core/src/main/java/com/oath/oak/MinKeyBuffer.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import sun.nio.ch.DirectBuffer;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+class MinKeyBuffer implements OakWriteBuffer, OakUnsafeDirectBuffer {
```

Comment:
The comment on the previous implementation of the min key buffer class is still relevant with the new code. Main problems:
1. The fact that the buffer is written in the time of its construction doesn't make it writable. It is read only buffer. I believe it should be OakAttachedReadBuffer/KeyBuffer.
2. It off-heap allocation on demand and it is costly. Either put on-heap or... Better to use OakAttachedReadBuffer/KeyBuffer and allocate via reused memory

Reply:
See comment about MinKeyBuffer above.